docs/path-resolution.rst

```diff
@@ -0,0 +1,690 @@
+.. _path-resolution:
+
+***************
+Path Resolution
+***************
+
+In order to be able to support reproducible builds on all platforms, the Solidity compiler has to
+abstract away the details of the filesystem where source files are stored.
+Paths used in imports must work the same way everywhere while the command-line interface must be
+able to work with platform-specific paths to provide good user experience.
+This section aims to explain in detail how Solidity reconciles these requirements.
+
+.. index:: ! virtual filesystem, ! source unit name
+.. _virtual-filesystem:
+
+Virtual Filesystem
+==================
+
+The paths used in :ref:`imports <import>` in a general case do not have to be filesystem paths.
+The compiler maintains an internal database (virtual filesystem) where each compiled source unit is
```

Comment:
Why "compiled source unit" and not just "source unit"?

Reply:
I did it mostly to avoid repetition. Changed.